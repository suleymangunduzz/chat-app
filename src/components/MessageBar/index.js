import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const MessageBar = ({
    sendMessage,
    handleMessageChange,
    sendMessageFromKeyboard,
    message
}) => {
    const [ keys, setKeys ] = useState([]);

    useEffect(() => {
        document.addEventListener('keydown', keyPressHandler);
        document.addEventListener('keyup', keyPressClean);

        return () => {
            window.removeEventListener('keydown', keyPressHandler);
            window.removeEventListener('keyup', keyPressClean);
        };
    }, []);

    useEffect(() => {
        if (keys.length >= 2 && keys[keys.length-2] === 'Meta' && keys[keys.length-1] === 'Enter' && sendMessageFromKeyboard) {
            sendMessage();
        }
    }, [keys]);

    const keyPressHandler = (event) => {
        setKeys(keys => [...keys, event.key])
    }

    const keyPressClean = () => {
        setKeys([]);
    }


    return (
        <div className="message-bar">
            <textarea className="message-bar__input" placeholder="Enter a message..." onChange={handleMessageChange} value={ message } />
            <button className="message-bar__button" onClick={ sendMessage }>Send</button>
        </div>
    );
}

MessageBar.propTypes = {
    sendMessage: PropTypes.func,
    handleMessageChange: PropTypes.func,
    message: PropTypes.string,
    sendMessageFromKeyboard: PropTypes.bool
}

export default MessageBar;
