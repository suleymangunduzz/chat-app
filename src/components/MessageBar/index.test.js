import React from 'react';
import renderer from 'react-test-renderer';
import MessageBar from './index';

const mockSendMessage = jest.fn();
const mockHandleMessageChange = jest.fn();

test('MessageBar component render', () => {
    
const mockSendMessageFromKeyboard = true;
const mockMessage = 'Mock message text.';

    const component = renderer.create(
        <MessageBar
            sendMessage={mockSendMessage}
            handleMessageChange={mockHandleMessageChange}
            sendMessageFromKeyboard={mockSendMessageFromKeyboard}
            message={mockMessage}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});