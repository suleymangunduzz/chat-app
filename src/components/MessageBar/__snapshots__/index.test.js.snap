// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBar component render 1`] = `
<div
  className="message-bar"
>
  <textarea
    className="message-bar__input"
    onChange={[MockFunction]}
    placeholder="Enter a message..."
    value="Mock message text."
  />
  <button
    className="message-bar__button"
    onClick={[MockFunction]}
  >
    Send
  </button>
</div>
`;
